### PR TITLE
Swift 6 compatibility

### PR DESCRIPTION
**Pull request by Supereg**
# Swift 6 compatibility

## :recycle: Current situation & Problem
This PR enables strict concurrency checking for SpeziAccount and adds compatibility with Swift 6.

## :gear: Release Notes 
* Enable strict concurrency checking.

## :books: Documentation
--

## :white_check_mark: Testing
--

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: Sources/SpeziAccount/AccountService/Configuration/UserIdType.swift [hunk 1]
```diff
@@ -10,7 +10,7 @@ import Foundation
 
 
 /// Determines the type and kind of the ``UserIdKey``.
-public enum UserIdType: Sendable, Equatable {
+public enum UserIdType {
```
Comment (PSchmiedmayer) [inline Sources/SpeziAccount/AccountService/Configuration/UserIdType.swift:13]: There seems to be an indentation problem here between line 17 & 18.

File: Sources/SpeziAccount/Resources/Localizable.xcstrings [hunk 2]
```diff
@@ -1126,6 +1126,7 @@
     },
     "Preview Service" : {
       "comment" : "No need to translate, only used in Previews ...",
+      "extractionState" : "stale",
```
Comment (PSchmiedmayer) [inline Sources/SpeziAccount/Resources/Localizable.xcstrings:1129]: Probably can be removed

File: Sources/SpeziAccount/Views/DataDisplay/LocalizableStringBasedDisplayView.swift [hunk 3]
```diff
@@ -30,7 +30,15 @@ public struct LocalizableStringBasedDisplayView<Key: AccountKey>: DataDisplayVie
 }
 
 
-extension Bool: CustomLocalizedStringResourceConvertible {
+#if compiler(<6)
```
Comment (PSchmiedmayer) [inline Sources/SpeziAccount/Views/DataDisplay/LocalizableStringBasedDisplayView.swift:33]: I think that should be the other way around.

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the improvements here @Supereg; only had very minor comments.

I assume from the API surface level this will be a minor release?

File: Sources/SpeziAccount/AccountService/Configuration/UserIdType.swift (hunk 1 above)
Comment (Supereg) [reply Sources/SpeziAccount/AccountService/Configuration/UserIdType.swift:13]: This seems to be a display issue within GitHub. If you count the spaces in the browser manually it matches even.

**Pull request merged**